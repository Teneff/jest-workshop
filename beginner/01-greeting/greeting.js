export default function greeting(name = 'Guest') {
  return `Grettings, ${name}!`;
}
