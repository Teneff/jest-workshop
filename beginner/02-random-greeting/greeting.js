export default function greeting(name = 'Guest') {
  const Greetings = [
    {
      phrase: `Good morning, ${name}!`,
      from: 5,
      to: 11,
    },
    {
      phrase: `Good afternoon, ${name}!`,
      from: 13,
      to: 17,
    },
    {
      phrase: `Good evening, ${name}!`,
      from: 17,
      to: 22,
    },
    {
      phrase: `Hello, ${name}!`,
      from: 0,
      to: 24,
    },
  ];
  const hours = new Date().getHours();
  const { phrase } = Greetings.find(
    ({ from, to }) => from <= hours && hours <= to,
  );
  return phrase;
}
