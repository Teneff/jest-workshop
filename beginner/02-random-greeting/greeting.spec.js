import greeting from './greeting';

describe('random greeting', () => {
  beforeAll(() => {
    jest.spyOn(Date.prototype, 'getHours');
  });

  describe.each([...Array(24).keys()])('at %d:00', (hours) => {
    beforeAll(() => {
      Date.prototype.getHours.mockReturnValue(hours);
    });

    describe('without given name', () => {
      it('should greet the guest', () => {
        expect(greeting()).toMatchSnapshot();
      });
    });

    describe('with given name', () => {
      it('should greet the user', () => {
        expect(greeting()).toMatchSnapshot();
      });
    });
  });
});
