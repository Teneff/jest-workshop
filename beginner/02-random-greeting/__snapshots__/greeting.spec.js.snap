// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`random greeting at 0:00 with given name should greet the user 1`] = `"Hello, Guest!"`;

exports[`random greeting at 0:00 without given name should greet the guest 1`] = `"Hello, Guest!"`;

exports[`random greeting at 1:00 with given name should greet the user 1`] = `"Hello, Guest!"`;

exports[`random greeting at 1:00 without given name should greet the guest 1`] = `"Hello, Guest!"`;

exports[`random greeting at 2:00 with given name should greet the user 1`] = `"Hello, Guest!"`;

exports[`random greeting at 2:00 without given name should greet the guest 1`] = `"Hello, Guest!"`;

exports[`random greeting at 3:00 with given name should greet the user 1`] = `"Hello, Guest!"`;

exports[`random greeting at 3:00 without given name should greet the guest 1`] = `"Hello, Guest!"`;

exports[`random greeting at 4:00 with given name should greet the user 1`] = `"Hello, Guest!"`;

exports[`random greeting at 4:00 without given name should greet the guest 1`] = `"Hello, Guest!"`;

exports[`random greeting at 5:00 with given name should greet the user 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 5:00 without given name should greet the guest 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 6:00 with given name should greet the user 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 6:00 without given name should greet the guest 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 7:00 with given name should greet the user 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 7:00 without given name should greet the guest 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 8:00 with given name should greet the user 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 8:00 without given name should greet the guest 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 9:00 with given name should greet the user 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 9:00 without given name should greet the guest 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 10:00 with given name should greet the user 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 10:00 without given name should greet the guest 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 11:00 with given name should greet the user 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 11:00 without given name should greet the guest 1`] = `"Good morning, Guest!"`;

exports[`random greeting at 12:00 with given name should greet the user 1`] = `"Hello, Guest!"`;

exports[`random greeting at 12:00 without given name should greet the guest 1`] = `"Hello, Guest!"`;

exports[`random greeting at 13:00 with given name should greet the user 1`] = `"Good afternoon, Guest!"`;

exports[`random greeting at 13:00 without given name should greet the guest 1`] = `"Good afternoon, Guest!"`;

exports[`random greeting at 14:00 with given name should greet the user 1`] = `"Good afternoon, Guest!"`;

exports[`random greeting at 14:00 without given name should greet the guest 1`] = `"Good afternoon, Guest!"`;

exports[`random greeting at 15:00 with given name should greet the user 1`] = `"Good afternoon, Guest!"`;

exports[`random greeting at 15:00 without given name should greet the guest 1`] = `"Good afternoon, Guest!"`;

exports[`random greeting at 16:00 with given name should greet the user 1`] = `"Hello, Guest!"`;

exports[`random greeting at 16:00 without given name should greet the guest 1`] = `"Hello, Guest!"`;

exports[`random greeting at 17:00 with given name should greet the user 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 17:00 without given name should greet the guest 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 18:00 with given name should greet the user 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 18:00 without given name should greet the guest 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 19:00 with given name should greet the user 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 19:00 without given name should greet the guest 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 20:00 with given name should greet the user 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 20:00 without given name should greet the guest 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 21:00 with given name should greet the user 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 21:00 without given name should greet the guest 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 22:00 with given name should greet the user 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 22:00 without given name should greet the guest 1`] = `"Good evening, Guest!"`;

exports[`random greeting at 23:00 with given name should greet the user 1`] = `"Hello, Guest!"`;

exports[`random greeting at 23:00 without given name should greet the guest 1`] = `"Hello, Guest!"`;
